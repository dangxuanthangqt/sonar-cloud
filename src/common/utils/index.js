/* eslint-disable no-bitwise */
/* eslint-disable func-names */
export const handleRegexWithSpecialCharacters = (string) =>
  string.replace(/([.?*+^$[\]\\(){}|-])/g, '\\$1');

export const formatHint = (string) => string.replace(/\n/gm, '<br/><br/>');

export const getDateFormatString = () => {
  const locale = window.navigator.userLanguage || window.navigator.language;
  const formatObj = new Intl.DateTimeFormat(locale).formatToParts(new Date());

  return formatObj
    .map((obj) => {
      switch (obj.type) {
        case 'day':
          return 'dd';
        case 'month':
          return 'MM';
        case 'year':
          return 'yyyy';
        default:
          return obj.value;
      }
    })
    .join('');
};

export const truncateString = (str = '', num = 100) => {
  if (str.length > num) {
    return `${str.slice(0, num)} ...`;
  }

  return str;
};

export function uuidv4() {
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
    const r = (Math.random() * 16) | 0;
    const v = c === 'x' ? r : (r & 0x3) | 0x8;
    return v.toString(16);
  });
}
