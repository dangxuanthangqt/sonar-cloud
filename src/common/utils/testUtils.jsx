import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { StyledEngineProvider } from '@mui/material';
import { ThemeProvider } from '@mui/material/styles';
import { muiTheme } from 'layouts/baseStyle';
import { QueryClient, QueryClientProvider } from 'react-query';
import { RecoilRoot } from 'recoil';
import { render as rtlRender } from '@testing-library/react';

function AllTheProviders({ children }) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnWindowFocus: false,
      },
    },
  });

  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={muiTheme}>
        <QueryClientProvider client={queryClient}>
          <RecoilRoot>
            <Router>{children}</Router>
          </RecoilRoot>
        </QueryClientProvider>
      </ThemeProvider>
    </StyledEngineProvider>
  );
}

const render = (ui, options) => {
  return rtlRender(ui, { wrapper: AllTheProviders, ...options });
};

export { render };
export * from '@testing-library/react';
