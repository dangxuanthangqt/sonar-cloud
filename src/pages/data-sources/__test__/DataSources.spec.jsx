import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, waitFor } from '../../../common/utils/testUtils';
import DataSources from '@/pages/data-sources';

describe('Data source', () => {
  it('should match snapshot', async () => {
    const { container } = render(<DataSources />);
    expect(container).toMatchSnapshot();
  });

  it('displays returned tasks on successful fetch', async () => {
    const { getByText, getByRole } = render(<DataSources />);

    expect(getByRole('button', { name: 'Next' })).toBeInTheDocument();

    expect(getByText('Datasources')).toBeInTheDocument();

    expect(getByText('Keywords')).toBeInTheDocument();

    expect(getByText('Vehicles')).toBeInTheDocument();

    expect(getByText('Sales Codes')).toBeInTheDocument();

    expect(getByText('LOPs Parts')).toBeInTheDocument();

    expect(getByText('Summary')).toBeInTheDocument();

    expect(getByText(/Step 1/)).toBeInTheDocument();

    // await waitFor(() =>
    //   expect(getByText(/Field Reports/i)).toBeInTheDocument()
    // );

    // await waitFor(() => {
    //   selectAllButton = getByRole('button', { name: 'buttons.select_all' });
    //   expect(selectAllButton).toBeInTheDocument();
    // });

    // const firstOption = getByRole('checkbox', {
    //   name: /straight time claim narrative \(stn\)/i,
    // });

    // expect(firstOption).toBeInTheDocument();
    // expect(firstOption).toHaveProperty('checked', false);

    // userEvent.click(selectAllButton);
    // expect(firstOption).toHaveProperty('checked', true);

    // const saveParamsButton = getByRole('button', {
    //   name: 'buttons.save_parameters',
    // });

    // userEvent.click(saveParamsButton);
    // await waitFor(() =>
    //   expect(
    //     getByRole('button', { name: 'buttons.in_progress' })
    //   ).toBeInTheDocument()
    // );

    // expect(saveParamsButton).toBeInTheDocument();
  });
});
