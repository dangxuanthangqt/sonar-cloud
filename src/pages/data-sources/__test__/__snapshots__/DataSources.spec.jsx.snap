// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data source should match snapshot 1`] = `
<div>
  <main>
    <div
      class="MuiBox-root css-v8fepg"
    >
      <div
        class="flex justify-center flex-col items-center"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-coieiy-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
            data-testid="NavigateBeforeRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.91 6.71a.9959.9959 0 0 0-1.41 0L8.91 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L11.03 12l3.88-3.88c.38-.39.38-1.03 0-1.41z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <p
          style="font-size: 14px; font-weight: 500; color: rgb(18, 41, 62); margin-top: 3px; margin-bottom: 0px; display: none;"
        >
          Previous
        </p>
      </div>
      <div
        class="MuiStepper-root MuiStepper-horizontal css-1enz6m5-MuiStepper-root"
      >
        <div
          class="MuiStep-root MuiStep-horizontal css-4u4pbf-MuiStep-root"
        >
          <span
            class="makeStyles-stepLabelRoot-8 MuiStepLabel-root MuiStepLabel-horizontal css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="makeStyles-iconStyle-9 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  1
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="makeStyles-labelStyle1-10 MuiStepLabel-label makeStyles-activeStep-11 Mui-active css-gv1taf-MuiStepLabel-label"
              >
                Datasources
              </span>
            </span>
          </span>
        </div>
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
          />
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal css-4u4pbf-MuiStep-root"
        >
          <span
            class="makeStyles-stepLabelRoot-8 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="makeStyles-iconStyle-9 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  2
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="makeStyles-labelStyle1-10 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
              >
                Keywords
              </span>
            </span>
          </span>
        </div>
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
          />
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal css-4u4pbf-MuiStep-root"
        >
          <span
            class="makeStyles-stepLabelRoot-8 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="makeStyles-iconStyle-9 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  3
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="makeStyles-labelStyle1-10 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
              >
                Vehicles
              </span>
            </span>
          </span>
        </div>
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
          />
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal css-4u4pbf-MuiStep-root"
        >
          <span
            class="makeStyles-stepLabelRoot-8 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="makeStyles-iconStyle-9 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  4
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="makeStyles-labelStyle1-10 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
              >
                Sales Codes
              </span>
            </span>
          </span>
        </div>
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
          />
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal css-4u4pbf-MuiStep-root"
        >
          <span
            class="makeStyles-stepLabelRoot-8 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="makeStyles-iconStyle-9 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  5
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="makeStyles-labelStyle1-10 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
              >
                Dates
              </span>
            </span>
          </span>
        </div>
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
          />
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal css-4u4pbf-MuiStep-root"
        >
          <span
            class="makeStyles-stepLabelRoot-8 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="makeStyles-iconStyle-9 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  6
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="makeStyles-labelStyle1-10 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
              >
                LOPs Parts
              </span>
            </span>
          </span>
        </div>
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
          />
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal css-4u4pbf-MuiStep-root"
        >
          <span
            class="makeStyles-stepLabelRoot-8 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="makeStyles-iconStyle-9 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  7
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="makeStyles-labelStyle1-10 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
              >
                Summary
              </span>
            </span>
          </span>
        </div>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ilvgj4-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
            data-testid="NavigateNextRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9.31 6.71c-.39.39-.39 1.02 0 1.41L13.19 12l-3.88 3.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41L10.72 6.7c-.38-.38-1.02-.38-1.41.01z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <p
          style="font-size: 14px; font-weight: 500; color: rgb(18, 41, 62); margin-top: 3px; margin-bottom: 0px;"
        >
          Next
        </p>
      </div>
    </div>
    <div
      class="makeStyles-contentContainer-3"
      id="123456"
    >
      <div
        class="makeStyles-content-5 makeStyles-contentWithCollapsedSidebar-6 makeStyles-contentWithStepper-7"
      >
        <div
          aria-hidden="true"
          class="MuiBackdrop-root css-1ji7omv-MuiBackdrop-root"
          role="loading"
          style="opacity: 0; visibility: hidden;"
        >
          <span
            class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </span>
        </div>
        <div
          style="margin-bottom: 25px;"
        >
          <div
            class="makeStyles-header1-21"
          >
            Create New Request
          </div>
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary makeStyles-input-19 css-ffkbon-MuiInputBase-root-MuiInput-root"
          >
            <input
              class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              inputmode="numeric"
              pattern="[0-9]*"
              type="text"
              value="This is request title This is request title"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-10hr09j"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-89t54g-MuiTypography-root"
            >
              Step 
              1
               :
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uucx26-MuiTypography-root"
            >
              Data Sources
            </p>
          </div>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1i3jxbp-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Review
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
        />
        <div
          class="MuiBox-root css-slhovo"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-7t5mk2-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CancelOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.59-13L12 10.59 8.41 7 7 8.41 10.59 12 7 15.59 8.41 17 12 13.41 15.59 17 17 15.59 13.41 12 17 8.41z"
                />
              </svg>
            </span>
            Cancel
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-t8y70b-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Next
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SkipNextRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m7.58 16.89 5.77-4.07c.56-.4.56-1.24 0-1.63L7.58 7.11C6.91 6.65 6 7.12 6 7.93v8.14c0 .81.91 1.28 1.58.82zM16 7v10c0 .55.45 1 1 1s1-.45 1-1V7c0-.55-.45-1-1-1s-1 .45-1 1z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </main>
</div>
`;
